core/go/internal/identityresolver/identityresolver.go

```diff
@@ -175,13 +175,33 @@ func (ir *identityResolver) ResolveVerifierAsync(ctx context.Context, lookup str
 			return
 		}
 
-		err = ir.transportManager.Send(ctx, &components.FireAndForgetMessageSend{
+		errChan := make(chan error, 1)
+		err = ir.transportManager.SendWithNack(ctx, &components.FireAndForgetMessageSend{
 			MessageID:   &requestID,
 			MessageType: "ResolveVerifierRequest",
 			Component:   prototk.PaladinMsg_IDENTITY_RESOLVER,
 			Node:        remoteNodeId,
 			Payload:     resolveVerifierRequestBytes,
-		})
+		}, errChan)
+
+		go func() {
+			// Handle transport-layer errors (e.g. the transport plugin couldn't reach the remote node)
+			// If sending fails, fail the request in the same manor we would if the remote node had returned an error response
+			select {
+			case <-ctx.Done():
+				return
+			case err, ok := <-errChan:
```

Comment:
I've not seen the pattern of "close means good". I'm overall just confused why this isn't a synchronous function `SendSync` - and we just call it in a go routine. The interface just feels half-formed to me.